File: aldous_broder_colored.rb
require 'colored_grid'
require 'aldous_broder'

6.times do |n|
  grid = ColoredGrid.new(20, 20)
  AldousBroder.on(grid)

  middle = grid[grid.rows/2, grid.columns/2]
  grid.distances = middle.distances

  filname = "aldous_broder_%02d.png" % n
  grid.to_png.save(filname)
  puts "saved #{filname}"
end

File: jmazes/src/AldousBroderColored.java
import util.ImageSaver;

import java.util.stream.IntStream;

public class AldousBroderColored {

    public static void main(String[] args) {

        IntStream.range(0, 6).
                forEach(n -> {
                    ColoredGrid grid = new ColoredGrid(20, 20);
                    AldousBroder.on(grid);

                    Cell middle = grid.get(grid.rows() / 2, grid.columns() / 2);
                    grid.setDistances(middle.distances());

                    String filename = String.format("jmazes/aldous_broder_%02d.png", n);
                    ImageSaver.saveImage(grid.toImage(), filename);
                    System.out.println("saved " + filename);
                });
    }
}
